### ISSUE TITLE
FNX2-15621 ⁃ Please don't release to stable channel before feature parity with Fennec!

### ISSUE THREAD

**Issue by opusforlife2** [mozilla-mobile/fenix#13022]
This is neither a feature request, nor a bug, but a meta issue, so this seemed like a catch all choice. I'm opening this issue because I've started catching snippets here and there that Fenix is about to be rolled out to the stable Firefox channel soon. I hope this is not the case, because it **isn't ready from a user's perspective**.

## User Story
As a user, I want Fenix to have all the features that Fennec currently has. Off the top of my head, that currently leaves pending:

* tab queues
* about:config support
* the ability to install all extensions which are supported on Fennec, not just Mozilla recommended ones.
* Support for view-source: (suggested by @nbmrjuhneibkr) (Fennec has it in the Page submenu of 3 dot menu)
* List of bugs marked for the transition that didn't make it through for some reason - https://github.com/mozilla-mobile/fenix/labels/Feature%3AFennecTransition (suggested by @justdave)

Maybe other users have more to list that I can't think of right now.

But the point is this: I, and I'm sure many other users, install and set up Firefox for other non-tech-savvy people, including the elderly. They often have use cases that are satisfied by 'that one extension' or 'that one pref in about:config' and nothing else. If they're automatically upgraded to Fenix and lose important extensions and settings, they'll be hung out to dry.

These are COVID times. Most of these people aren't currently reachable for us. Maybe some of them could manage to get it working again, but most would be left with a bad taste in their mouths regarding Firefox, leading to their installing Chrome or another alternative. This is a user base alienation issue as well. We know how most people are with breaking changes. There has been enough evidence of this gathered for many apps and services over the years. The transition should be as smooth as possible, and one of the fundamental pillars for this is feature parity.

Currently, tab queues (#470) is the only issue I see marked with the Fennec transition label, but it hasn't seen any activity recently. Please consider the other two bullet points as well. A delay in releasing would be noticed by a very small subset of users, a lot of whom are active participators here, but a botched release could make a long lasting bad impression on millions of users, according to Play Store install statistics.

### Acceptance Criteria
* I can do everything in Fenix's release version that I could do on Fennec, including building tab queues, changing prefs in about:config, and installing any and all extensions which are supported by Fennec.

**Comment by nbmrjuhneibkr** [comment]
Yeah, this release is going to be a disaster once it rolls out for everyone, and may result in a lot of users abandoning Firefox on Android altogether. Whenever Mozilla does something that is intended to grow the user base, the result ends up being the opposite.

By the way, here is another basic feature that is currently missing from Fenix: https://github.com/mozilla-mobile/fenix/issues/3972

**Comment by nbmrjuhneibkr** [comment]
As far as UI-related regressions go, one of the more noticeable ones would be the lack quick access to bookmarks/history from the home page. At least it's being worked on, apparently.
https://github.com/mozilla-mobile/fenix/issues/12271
https://github.com/mozilla-mobile/fenix/issues/12941

**Comment by opusforlife2** [comment]
Yeah, currently, I have to either access them from the History/Bookmarks menu, or tap on the address bar and enter a space to see them.

**Comment by JoGarWeb** [comment]
I'm really miss "tab queue" feature 😪

**Comment by dnebdal** [comment]
I use Firefox stable as my main (only, really) android browser, and I have to say I was surprised to do a normal software update and suddenly get a very different UI, several extensions disabled, and my quick launch list is empty. I had a vague idea that the team were up to some sort of rewrite, but that's still a really rough transition to dump on me a random Tuesday evening with no warning.

**Comment by nbmrjuhneibkr** [comment]
Very rough indeed. I was expecting an in-app update notification at least 1 month in advance, but instead I found out about the Fenix release from a Reddit post.

**Comment by justdave** [comment]
Here's a list of bugs that were designated as important to have a smooth transition from Fennec to Fenix which aren't fixed yet:  https://github.com/mozilla-mobile/fenix/labels/Feature%3AFennecTransition

**Comment by SergeiFranco** [comment]
I got forced this update and my experience is awful.
It is just is unpolished.
The sad part is as a end user I cannot roll back.

**Comment by opusforlife2** [comment]
Oh well... it's too late now, I guess. Sorry guys, I should have opened this issue earlier. Going by the release timing, the process had likely already started much before I opened it.

**Comment by opusforlife2** [comment]
And the negative press begins... https://www.xda-developers.com/mozilla-firefox-79-for-android-preview-nightly-update-without-extensions-add-ons/

**Comment by creesch** [comment]
Unfortunately as far as addons/extensions go they seem to have made up their minds months ago, see https://github.com/mozilla-mobile/fenix/issues/5315.

**Comment by cadeyrn** [comment]
In #5315 was said that Mozilla wants to expand the add-on support in the future, so I suggest to read it again so that you don't have to write the same in more issues (it's already the third).

**Comment by creesch** [comment]
> In #5315 it was said that Mozilla wants to expand the add-on support in the future, so I suggest to read it again 


No need to be snippy. If you read it they never commit to allowing anything else than the extensions they want to allow. In fact "expanding add-on" support is a very nice way to not commit to anything at all. 

[This comment](https://github.com/mozilla-mobile/fenix/issues/5315#issuecomment-582953593) and the resulting discussion resulting [in this statement](https://github.com/mozilla-mobile/fenix/issues/5315#issuecomment-583710384) clearly indicates to me that the future status of installing anything besides recommended extensions is very much unclear as they clearly don't want to commit to simply allowing people to install extensions as they can now. 

Something that was also pointed out to you in that very same thread when you linked to a FAQ blog as if that answered the question back then, which it doesn't. The simple fact is that so far they haven't given a straight answer at all about what extension support will look like for anything besides recommended extensions. 

> so that you don't have to write the same in more issues (it's already the third).

There are a few relevant open issues where I figured this information is useful for the people in there. I don't see why I shouldn't share the same information there. 
A lot of the things discussed over 5 months ago are clearly relevant as there are multiple related issues open now.

**Comment by kbrosnan** [comment]
The Firefox for Android release migration has started. The main devs of this product are a team of about 6-10 people, supported by the Android Components and GeckoView teams which are similarly sized but support more than just Fenix. While that sounds like a lot of people it is a lot smaller than our main competitors. 

Fenix is a clean slate development of the Firefox Android browser. This gives us space to rethink and re-evaluate the features that were in Fennec. At this point the label [Feature:FennecTransition](https://github.com/mozilla-mobile/fenix/issues?q=is%3Aopen+is%3Aissue+label%3AFeature%3AFennecTransition) exists to document features that are in Fennec, not a commitment to build that exact same feature in Fenix. I liked tab queue too, being able to process my mail without context switching was great. Maybe it could come back as part of collections? A significant amount of development time was spent working on writing and validating the migration in H1 of 2020. Going into the rest of 2020 we are planning on more consistent feature development. Recently on Nightly users have seen some of this work. For example the long press back button for history stack and swipe on the nav bar to switch tabs. Other major work includes resolving the new tab/home screen flow complexities. We look forward to getting those and other features to the release users in the next few months if not sooner. 

To discuss this with us there are channels on [Element/Matrix](https://wiki.mozilla.org/Matrix) https://chat.mozilla.org/#/room/#fx-android-community:mozilla.org and #fenix along with the newly minted [Android Discourse](https://discourse.mozilla.org/c/firefox-android/468) community.

**Comment by filips123** [comment]
In this case, wouldn't it be better to wait before starting migration until some most commonly used features are implemented? I know Firefox ESR 68 has reached EOL, but how much effort would you need to maintain and keep security fixes for few more months?

I think Fenix has great potential to become very good browser. However, lack of some common or more advanced features might make users to stop updating Firefox or switch to alternative browsers.

**Comment by opusforlife2** [comment]
@kbrosnan
> A significant amount of development time was spent working on writing and validating the migration in H1 of 2020.

Ah. That means this discussion should have taken place some time in 2019. I don't remember there being any poll or community discussion regarding the features that would make the final cut back then. I'm sure your team took as much as possible into account when making the decision to set the release date, but it would have helped both the developers and users immensely if the various milestones for the Final Migration were discussed and communicated more clearly. It might have helped prioritise developer time better, or it might have brought a larger influx of contributions from community members for highly wanted features.

**Comment by opusforlife2** [comment]
By the way, for the users watching this thread whose countries haven't yet received the stable Fenix update: **mitigation.** Take advantage of the gradual rollout pattern. :)

I've informed as many people as I remember installing stable Firefox for, to not accept an update for stable Firefox on Android, and to disable auto update for it in case it is enabled. I'll be informing them about a good time to update at a later date, whenever the needed functionality lands. At least this way, they wouldn't be hampered by missing expected basic features, like certain extensions, a Downloads menu, and so on.

**Comment by justdave** [comment]
Agreed. I know I won't be updating it yet. I have nightly Fenix installed and use it as my primary browser and love it. But I have stable Firefox (Fennec) installed alongside it and still have to use it occasionally because of features Fenix doesn't support yet. Fennec will be avoiding updates until Fenix gets those.

**Comment by opusforlife2** [comment]
Go away, you bot.
